eng/Versions.props

```diff
@@ -157,7 +157,7 @@
     <RoslynOptProfRunSettingsGeneratorVersion>1.0.0-beta3.19057.1</RoslynOptProfRunSettingsGeneratorVersion>
     <RoslynToolsLightUpSystemRuntimeLoaderFixedVersion>4.3.0</RoslynToolsLightUpSystemRuntimeLoaderFixedVersion>
     <RoslynMicrosoftVisualStudioExtensionManagerVersion>0.0.4</RoslynMicrosoftVisualStudioExtensionManagerVersion>
-    <StreamJsonRpcVersion>1.3.23</StreamJsonRpcVersion>
+    <StreamJsonRpcVersion>2.0.106-beta</StreamJsonRpcVersion>
```

Comment:
Why are you depending on 2.0.106-beta here rather than the stable 2.0 build on nuget.org?

Reply:
@AArnott I am using the same version as test explorer team (@shyamnamboodiripad) 
since at run time, we will pick up whatever 2.x version in VS, I don't concern much on that, I am more concern on compile time and warnings from nuget. so I am trying to use same version with people who will depend on Roslyn.

Reply:
I don't think we should build a huge nuget package structure with the idea of all using exactly the same version of a given package, especially when it encourages using unstable packages where stable alternatives exist. In particular, this version is *older* than the one in VS. You already are straddling multiple versions and having to resolve it with explicit PackageReferences in a few projects, which is fine. So I encourage you to go ahead and use a stable version. That also means that your own downstream users can ship stable nuget packages. As you have it now, your package and downstream packages will have to be unstable since they depend on an unstable package.

Reply:
We should use the same version as is shipping in VS.

Reply:
VS's default.config file often refers to a slightly older "unstable" package version because inserting to VS usually precedes the stable package release (in case issues are caught that require more changes to the package). So you can choose the first stable package version on nuget.org that follows the version in VS if VS uses an unstable package.

But FWIW, we don't add/remove any public API within a major.minor release, so any 2.0 release on nuget.org should be adequate for compiling against. For testing fidelity with VS, choosing a release that's close to what VS ships is worth considering.

Reply:
@heejaechang Tried to update to StreamJsonRpc to 2.0.140 (or 146), but that version uses newer System.Memory (4.5.2) than we insert into VS. We insert 4.5.1. See https://dev.azure.com/devdiv/DevDiv/_git/VS?path=%2F.corext%2FConfigs%2Fdefault.config&version=GBrel%2Fd16.0.

StreamJsonRpc should not be referencing newer dependencies than those in VS. We can indeed update System.Memory we insert to 4.5.2, but that's better done in a separate PR.

Reply:
StreamJsonRpc [upgraded to System.Memory 4.5.2](https://github.com/Microsoft/vs-streamjsonrpc/commit/303df14e7e948c8c2cff09c3749a7a11a3db4e87) because it contained a package metadata fix. The assembly version within the package is _unchanged_ so this isn't a hazardous situation for VS and does not require another System.Memory insertion. 

The System.Memory package version 4.5.2 was necessary for StreamJsonRpc to take because without it, StreamJsonRpc was broken by default since binding redirects (for System.Buffers IIRC) would be missing in the build-generated .config file.

Reply:
Ok. But now we have an issue with NuGet in Roslyn because we explicitly depend on NuGet version 4.5.1 and StreamJsonRpc  is bringing in NuGet version 4.5.2. So we will need to upgrade what we insert into VS to avoid that conflict.

Reply:
I tried to move to newer version and moved back to the beta as I am using now. I am not going to move to newer version since that cause other set of dependency problem since newer version uses newer version of depdencies such as System.Memory and VS.Threading and etc which conflict with versions Roslyn is using. it starts new set of dependency hell.

I will move it to newer versino when Roslyn as whole move to newer version of dependencies.

this is only for build time, on runtime, we will pick up whatever shipped with VS.

Reply:
As you're referencing an older beta than is in VS, there is a risk that there are actually API breaking changes between what you're compiling against and what you have at runtime. In which case you'll suffer runtime failures in VS that you won't see in unit tests. If you proceed with this plan, you may want to diff the API between the two DLLs to assess the level of this risk.

Reply:
hmmm.. @shyamnamboodiripad according to @AArnott there is breaking change between 2.x in VS and this version. have you encountered issue? I didn't encounter anything in Roslyn side so far.

Reply:
Nah, I didn't say there *were* breaking changes between those two versions. I was merely pointing out that since the one you're referencing is a beta, that there _could_ be breaking changes. I just diff'd the two versions though (v2.0.106-beta..v2.0.123-beta), and changes are only _additive_, so you don't have a problem here.